File: api/Relations/MonthWeekRelation.cs
using api.DTOs;

namespace api.Relations
{
    public class MonthWeekRelation
    {
        public MonthWeekDTO MonthWeekDTO { get; set; }
        public virtual List<PlantDTO> SewedPlants { get; set; }
        public virtual List<PlantDTO> HarvestedPlants { get; set; }

        public virtual List<GardeningTaskDTO> GardeningTasks { get; set; }

        public MonthWeekRelation()
        {
            MonthWeekDTO = new MonthWeekDTO();
            SewedPlants = new List<PlantDTO>();
            HarvestedPlants = new List<PlantDTO>();
            GardeningTasks = new List<GardeningTaskDTO>();
        }
    }
}

File: client/src/models/MonthWeekRelation.ts
import { GardeningTaskDTO } from "./GardeningTaskDTO";
import { MonthWeekDTO } from "./MonthWeekDTO";
import { PlantDTO } from "./PlantDTO";

export interface MonthWeekRelation {
  monthWeekDTO: MonthWeekDTO;
  sewedPlants: Array<PlantDTO>;
  harvestedPlants: Array<PlantDTO>;
  gardeningTasks: Array<GardeningTaskDTO>;
}
